src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -279,6 +301,48 @@ private static void OnRequestSendingRequest(WinHttpRequestState state)
                 var serverCertificate = new X509Certificate2(certHandle);
                 Interop.Crypt32.CertFreeCertificateContext(certHandle);
 
+                IPAddress? ipAddress = null;
+                unsafe
+                {
+                    Interop.WinHttp.WINHTTP_CONNECTION_INFO connectionInfo;
+                    Interop.WinHttp.WINHTTP_CONNECTION_INFO* pConnectionInfo = &connectionInfo;
+                    uint infoSize = (uint)sizeof(Interop.WinHttp.WINHTTP_CONNECTION_INFO);
+                    if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(state, $"sizeof(WINHTTP_CONNECTION_INFO)={infoSize}");
+                    if (Interop.WinHttp.WinHttpQueryOption(
+                        state.RequestHandle,
+                        Interop.WinHttp.WINHTTP_OPTION_CONNECTION_INFO,
+                        (IntPtr)pConnectionInfo,
+                        ref infoSize))
+                    {
+                        ReadOnlySpan<byte> RemoteAddressSpan = new ReadOnlySpan<byte>(connectionInfo.RemoteAddress, 128);
+                        AddressFamily addressFamily = (AddressFamily)BitConverter.ToInt16(RemoteAddressSpan.ToArray(), 0);
```

Comment:
```suggestion
                        AddressFamily addressFamily = (AddressFamily)BitConverter.ToInt16(RemoteAddressSpan);
```

Reply:
Oh I guess this is due to Framework? :/

Reply:
Yes

Reply:
In that case we can at least do `.AsSpan(0, 2).ToArray()` instead to avoid allocating the full 128 bytes

Reply:
Can it be just `Unsafe.ReadUnaligned<short>(ref MemoryMarshal.GetReference(remoteAddressSpan))`? (Assuming we don't need to also check endianness here.)

Reply:
The important thing is to avoid unnecessary allocations, it doesn't really matter how. (The code I quoted is `BinaryPrimitives.ReadInt16LittleEndian`'s implementation.)